src/github/client/default-key-locator.ts

```diff
@@ -0,0 +1,23 @@
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { InstallationId } from "~/src/github/client/installation-id";
+import { GitHubServerApp } from "~/src/models/github-server-app";
+import { Subscription } from "~/src/models/subscription";
+import * as PrivateKey from "probot/lib/private-key";
+
+/**
+ * Look for a Github app's private key
+ */
+export const defaultKeyLocator = async (gitHubInstallation: InstallationId) => {
+	const subscription = await Subscription.findOneForGitHubInstallationId(gitHubInstallation.installationId);
```

Comment:
installationId is only unique per server, conflicts are possible.  You need to filter based on `appId` and `githubInstallationId`.  For cloud, that means that `appId` is NULL

Reply:
this scenario should also be added to the tests (same installation id, different appid)

Reply:
wait, actually, isn't it all based on the appId? you don't even need installation id...

Reply:
I think I see what Harminder is doing, the `installationId` is needed for the `jiraHost` so that we can do a jiraHost base FF here. 

I feel not Harminder's fault, but the code of how we use jiraHost base FF is really complicated. Need to pass it all the way down from req to bottom. So using `installationId` and go through the db might be "easier". <-- but good point @mboudreau , I don't think we should go this route. 

Let's think about another way while we can still use jiraHost base FF.

Reply:
I'm not sure an FF is needed frankly.  All we're doing is replicating the probot functionality with an additional step if the appId is specified for GHE, which won't be passed in if GHE_SERVER FF is not already enabled.  With some thorough unit testing, we should have enough confidence.

Reply:
Github `AppId ` is not unique across GHE instances.

Reply:
no, but githubAppId in our DB is :)